### ISSUE TITLE
Azurite holds a lock on the two directories `__blobstorage__` and `__queuestorage__`

### ISSUE THREAD

**Issue by danielniccoli** [Azure/Azurite#2356]
### Which service(blob, file, queue, table) does this issue concern?
blob, queue

### Which version of the Azurite was used?
v3.29.0

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)
Visual Studio Code Extension

### What problem was encountered?
After starting and stopping Azurite in VSCode using the `>Azurite: Start` and `>Azurite: Close` commands, The extension still holds a lock on the two directories `__blobstorage__` and `__queuestorage__`. Consequently, I am unable to delete these directories, for example to start anew, until I close VS Code. This makes that process unnecessary frustrating.

### Steps to reproduce the issue?
1. `>Azurite: Start`
2. `>Azurite: Close` 
3. Try deleting `__blobstorage__` and `__queuestorage__` in the project root. Will fail.
4. Close VS Code.
5. Try deleting `__blobstorage__` and `__queuestorage__` in the project root. Will succeed.

### Have you found a mitigation/solution?
No.

**Comment by blueww** [comment]
@danielniccoli 

Thanks for raising this issue!

In VS Code, Azurite is running in VS code process, so close Azurite will close the http service but can't stop whole process.

When Azurite read/save content in local disk, it will create Stream like https://github.com/Azure/Azurite/blob/d544d16f910e490fdd9db5565459df701895308f/src/common/persistence/FSExtentStore.ts#L343. 
When create Stream Azurite will not set autoClose , and the autoClose  default value is true (See [link ](https://nodejs.org/api/fs.html#fscreatereadstreampath-options)). So Azurite depends on NodeJS  to close the stream after finish using it. But currently it looks the stream is not closed and cause the related folder can't be deleted.

If you would like to have the content always be cleaned up after close Azurite, you can try to save the content in memory by enable following setting.
<img width="793" alt="image" src="https://github.com/Azure/Azurite/assets/6926193/58e22b0d-b784-4a53-994d-0baaf83607cd">

**Comment by blueww** [comment]
There are still not easy way to close the stream manually without changing the code, until we check all place to read the stream, and add code to close it after the last time using it. However, this might have regression risk, and we have other feature parity in hand, so this might won't be our recent priority.

**Comment by danielniccoli** [comment]
It may just be enough to close the stream when issuing the Azurite Close command.

**Comment by blueww** [comment]
@EmmaZhu 

Will your in progress PR https://github.com/Azure/Azurite/pull/2331 can help fix this issue?

**Comment by EmmaZhu** [comment]
@blueww , 

Yes, still working on it, will need to resolve some test failures.
